httpevents/handler.go

```diff
@@ -68,11 +69,83 @@ func NewHandlerWith(logger *events.Logger, handler http.Handler) http.Handler {
 	})
 }
 
+type LoggerFunc func(headers http.Header) http.Header
+
+func copyHeaders(headers http.Header) http.Header {
+	fmt.Println("Copying headers")
+	fmt.Println("Existing Header")
+	for k, v := range headers {
+		fmt.Println(k)
+		fmt.Println(v)
+	}
+	headersCopy := make(http.Header)
+
+	for k, v := range headers {
+		headersCopy[k] = v
+	}
+
+	fmt.Println("New Headers")
+	for k, v := range headersCopy {
+		fmt.Println(k)
+		fmt.Println(v)
+	}
+
+	return headersCopy
+}
+func NewHandlerWithFormatting(formatter LoggerFunc, logger *events.Logger, handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		var laddr string
+
+		if value, ok := req.Context().Value(http.LocalAddrContextKey).(net.Addr); ok {
+			laddr = value.String()
+		}
+
+		w := responseWriterPool.Get().(*responseWriter)
+		// We capture all the values we need from req in case the object
+		// gets modified by the handler.
+		w.ResponseWriter = res
+		w.logger = logger
+		w.SanitizeHeaders = formatter
```

Comment:
other than this line, i i just copy pasted the NewHandlerWith (can re-evaluate once we get things working)